ipa-core/src/protocol/context/dzkp_validator.rs

```diff
@@ -375,20 +310,19 @@ impl UnverifiedValuesStore {
         // segments are small, pack one or more in each entry of `vec`
         let position_bit_array = (length * position_raw) % 256;
 
-        // get entry
-        let entry = self.vec[position_vec].get_or_insert_with(UnverifiedValues::default);
+        let unverified_values = &mut self.vec[position_vec];
 
         // copy segment value into entry
         for (segment_value, array_value) in [
-            (segment.x_left, &mut entry.x_left),
-            (segment.x_right, &mut entry.x_right),
-            (segment.y_left, &mut entry.y_left),
-            (segment.y_right, &mut entry.y_right),
-            (segment.prss_left, &mut entry.prss_left),
-            (segment.prss_right, &mut entry.prss_right),
-            (segment.z_right, &mut entry.z_right),
+            (segment.x_left, &mut unverified_values.x_left),
+            (segment.x_right, &mut unverified_values.x_right),
+            (segment.y_left, &mut unverified_values.y_left),
+            (segment.y_right, &mut unverified_values.y_right),
+            (segment.prss_left, &mut unverified_values.prss_left),
+            (segment.prss_right, &mut unverified_values.prss_right),
+            (segment.z_right, &mut unverified_values.z_right),
         ] {
-            // unwrap safe since index are guaranteed to be within bounds
+            // panics when out of bounds
             let values_in_array = array_value
                 .get_mut(position_bit_array..position_bit_array + length)
```

Comment:
It is unclear from the parameter names what `position_raw` and `length` mean. Please use more descriptive names. Are these positions within the `UnverifiedValues` or within the `Segment` or within the larger vector of `UnverifiedValues`? Length of what? of the segment? If so, why not just call `segment.length()`?